akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -173,6 +173,12 @@ final class ActorTestKit private[akka] (val name: String, val config: Config, se
   def spawn[T](behavior: Behavior[T], name: String, props: Props): ActorRef[T] =
     Await.result(internalSystem ? (ActorTestKitGuardian.SpawnActor(name, behavior, _, props)), timeout.duration)
 
+  /**
+   * Stop the actor under test. To make sure it stopped, use [[TestProbe#expectTerminated]] method.
+   */
+  def stop[T](ref: ActorRef[T]): Unit =
+    Await.result(internalSystem ? { x: ActorRef[ActorTestKitGuardian.Ack.type] ⇒ ActorTestKitGuardian.StopActor(ref, x) }, timeout.duration)
```

Comment:
I'm skeptic to this, shouldn't the actor under test rather have a message for stopping it if it should be possible to stop it from the outside? If not, what is the use case when it is needed?

Reply:
Not all actors will have a message to stop them. Actually I think it is an omission in typed actors not to have something like `PoisonPill`.  It is needed, when the same actor is spawned with different configurations in several tests in a row under the same name to avoid the name clash.

Reply:
If they are supposed to be possible to stop from anywhere else than their spawning parent (and maybe also in that case) they should have an explicit message for it, this is a very conscious design decision in typed. I'd argue this for untyped as well, even if `PoisonPill` exists over there.

Anyways, the use case is good enough for me for the test kit, which is a different story than the actual actor behaviors since we have control over the parent.